shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala

```diff
@@ -937,6 +956,31 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
     }
   }
 
+  private def getStatsFromBlockMeta(blockmetas: List[BlockMetaData]): InternalRow = {
+    case class Holder(var count: Int = 0, var size: Long = 0)
+    // assuming all columns are in the same order within all blocks and have the same number of cols
+    val stats = blockmetas(0).getColumns.asScala.map { c =>
+      (Statistics.createStats(c.getPrimitiveType), Holder())
+    }
+    val cols = blockmetas(0).getColumns.size()
+    var j = 0
+    while (j < cols) {
+      var i = 0
+      while (i < blockmetas.size) {
+        val columnChunkMetaData = blockmetas(i).getColumns.get(j)
+        stats(j)._1.mergeStatistics(columnChunkMetaData.getStatistics)
+        stats(j)._2.count += columnChunkMetaData.getValueCount.toInt
+        stats(j)._2.size += columnChunkMetaData.getTotalSize
+        i += 1
+      }
+      j += 1
+    }
+    InternalRow.fromSeq(stats.flatMap { stat =>
+      List(stat._1.genericGetMin(), stat._1.genericGetMax(), stat._1.getNumNulls.toInt,
```

Comment:
What about for structs and things, like a CalendarInterval.  What does genericGertMin or gernericGetMax return in those cases? For Spark they are null.

Reply:
From what I saw, the structs meta gets written decomposed into child columns, e.g. if a struct has int and long columns, parquet will have a meta for each of the child columns. 
This is meta from a struct col that I ran

```
scala> df.printSchema
root
 |-- struct: struct (nullable = true)
 |    |-- number: integer (nullable = true)
 |    |-- str_number: string (nullable = true)
 |    |-- child_struct: struct (nullable = true)
 |    |    |-- child_field: string (nullable = true)


scala> df.show
+----------------+
|          struct|
+----------------+
|{1, a, {child0}}|
|{2, d, {child1}}|
|{3, f, {child2}}|
+----------------+

> docker run -it -v $PWD:/cudf nathanhowell/parquet-tools meta /cudf/gpu_struct_2/part-00000-3c218e57-41e8-4868-8a27-66bce7d54cc3-c000.snappy.parquet
file:          file:/cudf/gpu_struct_2/part-00000-3c218e57-41e8-4868-8a27-66bce7d54cc3-c000.snappy.parquet 
creator:       parquet-mr version 1.10.1 (build a89df8f9932b6ef6633d06069e50c9b7970bebd1) 
extra:         org.apache.spark.version = 3.1.1 
extra:         org.apache.spark.sql.parquet.row.metadata = {"type":"struct","fields":[{"name":"struct","type":{"type":"struct","fields":[{"name":"number","type":"integer","nullable":true,"metadata":{}},{"name":"str_number","type":"string","nullable":true,"metadata":{}},{"name":"child_struct","type":{"type":"struct","fields":[{"name":"child_field","type":"string","nullable":true,"metadata":{}}]},"nullable":true,"metadata":{}}]},"nullable":true,"metadata":{}}]} 

file schema:   spark_schema 
--------------------------------------------------------------------------------
struct:        OPTIONAL F:3 
.number:       OPTIONAL INT32 R:0 D:2
.str_number:   OPTIONAL BINARY L:STRING R:0 D:2
.child_struct: OPTIONAL F:1 
..child_field: OPTIONAL BINARY L:STRING R:0 D:3

row group 1:   RC:3 TS:187 OFFSET:4 
--------------------------------------------------------------------------------
struct:        
.number:        INT32 SNAPPY DO:0 FPO:4 SZ:66/64/0.97 VC:3 ENC:BIT_PACKED,PLAIN,RLE ST:[min: 1, max: 3, num_nulls: 0]
.str_number:    BINARY SNAPPY DO:0 FPO:70 SZ:51/49/0.96 VC:3 ENC:BIT_PACKED,PLAIN,RLE ST:[min: a, max: f, num_nulls: 0]
.child_struct: 
..child_field:  BINARY SNAPPY DO:0 FPO:121 SZ:70/74/1.06 VC:3 ENC:BIT_PACKED,PLAIN,RLE ST:[min: child0, max: child2, num_nulls: 0]
> docker run -it -v $PWD:/cudf nathanhowell/parquet-tools meta /cudf/cpu_struct_2/part-00000-189c2034-a376-4518-8928-d46de5e4ea65-c000.snappy.parquet
file:          file:/cudf/cpu_struct_2/part-00000-189c2034-a376-4518-8928-d46de5e4ea65-c000.snappy.parquet 
creator:       parquet-mr version 1.10.1 (build a89df8f9932b6ef6633d06069e50c9b7970bebd1) 
extra:         org.apache.spark.version = 3.1.1 
extra:         org.apache.spark.sql.parquet.row.metadata = {"type":"struct","fields":[{"name":"struct","type":{"type":"struct","fields":[{"name":"number","type":"integer","nullable":true,"metadata":{}},{"name":"str_number","type":"string","nullable":true,"metadata":{}},{"name":"child_struct","type":{"type":"struct","fields":[{"name":"child_field","type":"string","nullable":true,"metadata":{}}]},"nullable":true,"metadata":{}}]},"nullable":true,"metadata":{}}]} 

file schema:   spark_schema 
--------------------------------------------------------------------------------
struct:        OPTIONAL F:3 
.number:       OPTIONAL INT32 R:0 D:2
.str_number:   OPTIONAL BINARY L:STRING R:0 D:2
.child_struct: OPTIONAL F:1 
..child_field: OPTIONAL BINARY L:STRING R:0 D:3

row group 1:   RC:3 TS:187 OFFSET:4 
--------------------------------------------------------------------------------
struct:        
.number:        INT32 SNAPPY DO:0 FPO:4 SZ:66/64/0.97 VC:3 ENC:BIT_PACKED,PLAIN,RLE ST:[min: 1, max: 3, num_nulls: 0]
.str_number:    BINARY SNAPPY DO:0 FPO:70 SZ:51/49/0.96 VC:3 ENC:BIT_PACKED,PLAIN,RLE ST:[min: a, max: f, num_nulls: 0]
.child_struct: 
..child_field:  BINARY SNAPPY DO:0 FPO:121 SZ:70/74/1.06 VC:3 ENC:BIT_PACKED,PLAIN,RLE ST:[min: child0, max: child2, num_nulls: 0]

```

Reply:
The point is does this work? have we tested it? If it does, then it would be good to document that it is doing what Spark expects (puts in a null because Spark does not know how to use them for filtering).